cleanlab/outlier.py

```diff
@@ -0,0 +1,198 @@
+# Copyright (C) 2017-2022  Cleanlab Inc.
+# This file is part of cleanlab.
+#
+# cleanlab is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# cleanlab is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with cleanlab.  If not, see <https://www.gnu.org/licenses/>.
+
+"""
+Class to identify which examples are out of distribution.
+"""
+import warnings
+from typing import Optional
+import numpy as np
+from cleanlab.rank import get_ood_scores, get_outlier_scores
+
+
+class OutOfDistribution:
+    OUTLIER_PARAMS = tuple(("k", "t"))
+    OOD_PARAMS = tuple(("labels", "adjust_pred_probs", "method"))
```

Comment:
Why do you think these should be tuples instead of list or set?

Reply:
Set is best